### ISSUE TITLE
IAM Policies for Integrated Services

### ISSUE THREAD

**Issue by dpwrussell** [serverless-operations/serverless-step-functions#168]
# This is a Feature Proposal

## Description

Now that step functions have been enhanced with various integrated services, it seems like it might be a good idea to offer enhanced capabilities for creating the IAM roles necessary for those integrations to function correctly.

https://docs.aws.amazon.com/step-functions/latest/dg/connectors-iam-templates.html

For example, in order for the AWS Batch service integration to create correctly, it is necessary for the state machine execution role (that is generated automatically by `serverless-step-functions`) to have certain permissions as detailed here: https://docs.aws.amazon.com/step-functions/latest/dg/batch-iam.html

The AWS console auto-detects which permissions are needed when creating a step function, but I think that would be too complex here. Instead I think it would be nice if (much like the lambda execution role in base serverless) we could configure the state machine execution role.

### Example: Migrating from using lambda functions to manage AWS batch submissions, to using the AWS Batch service integration:

This used to be a stanza in my lambda execution role, but now it needs to go in the statemachine execution role instead, but I have no way to do that other than pre-creating the role and then referencing it as detailed here: https://www.npmjs.com/package/serverless-step-functions#iam-role

```yaml
    - Effect: Allow
      Action:
        - batch:DescribeJobs
        - batch:SubmitJob
      Resource: "*"
```

Basically I think it would be nice to be able to inline the statemachine role specifications instead of specifying an external role. Per state machine would probably be ideal. If someone wanted to use the same role for many state machines then perhaps it could alternately be specified at a higher level, I have not thought too much about that use-case.
